### ISSUE TITLE
To check if an element exists or to skip it if the element doesn't exist using flutter drive

### ISSUE THREAD

**Issue by deonchimple** [flutter/flutter#15852]
## Steps to Reproduce
Using flutter_driver in the QA environment. Options feels quite limited currently. I am trying to check if an element exists or not, perhaps a function on the like of finder.IsEmpty or something like IfExists in Appium which the skips the element if it doesn't exist and doesn't necessarily throw an exception.

## Logs
Application runs fine. Just looking for more functionalities of flutter driver.

## Flutter Doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel beta, v0.1.5, on Microsoft Windows [Version 6.1.7601], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[!] Android Studio (version 2.3)
    X Unable to find bundled Java version.
[√] Android Studio (version 3.0)
[√] VS Code (version 1.21.1)
[√] Connected devices (1 available)

! Doctor found issues in 1 category.

**Comment by zoechi** [comment]
What about
https://docs.flutter.io/flutter/flutter_driver/FlutterDriver/waitFor.html

See also 
-
 https://github.com/flutter/flutter/blob/466d15433fb4bbc013eca0d6819ad903627987b2/packages/flutter_driver/lib/src/driver/driver.dart#L748
- https://github.com/flutter/flutter/blob/e27bcd0f9dfa3f811edc669e438fdcb812a9fe27/dev/integration_tests/ui/test_driver/driver_test.dart#L24

**Comment by deonchimple** [comment]
@zoechi Unfortunately all this does is wait for the element but doesn't necessarily check if the element is visible/ present (Ideally returning a boolean value). 

For instance when I am searching for an element, I would like to give a condition which would go to the next element if the current element isn't found. This requires a boolean function like Ispresent, IsEmpty, isVisible available in Appium and selenium.

**Comment by jonsamwell** [comment]
@deonchimple can you not assume that if the wait times out that the element is not in the widget tree?  If so you could do something like this.

```
  Future<void> testStep() async {
    final isOpen = await isPresent(find.byType("Drawer"), world.driver);
    if (isOpen) {
      ...
    }
  }

  Future<bool> isPresent(SerializableFinder finder, FlutterDriver driver, {Duration timeout = const Duration(seconds: 1)}) async {
    try {
      await driver.waitFor(finder, timeout: timeout);
      return true;
    } catch (e) {
      return false;
    }
  }
```

**Comment by brianegan** [comment]
@jonsamwell Yep! I've been doing something almost exactly like that, but agree it would be great if that was a method that already existed, since i can imagine a lot of us have that same code in our code bases :)

**Comment by akaGelo** [comment]
````dart      await driver.waitForAbsent(loginButton);
````

**Comment by akaGelo** [comment]
https://github.com/flutter/flutter/blob/a817b4e6a4e80deefb88373d2a548b97c90d19c4/packages/flutter_driver/lib/src/driver/driver.dart#L491

````dart
      await driver.waitForAbsent(loginButton);
````

**Comment by Hixie** [comment]
Being able to do the equivalent of "Finder.evaluate()" in a driver test would make sense.

**Comment by kutlaykural** [comment]
But using that function;
```
Future<bool> isPresent(SerializableFinder finder, FlutterDriver driver, {Duration timeout = const Duration(seconds: 1)}) async {
    try {
      await driver.waitFor(finder, timeout: timeout);
      return true;
    } catch (e) {
      return false;
    }
  }
```
causes an error log like that
`VMServiceFlutterDriver: waitFor message is taking a long time to complete...`
is there a way to get rid of that error message?

**Comment by Akkasim46** [comment]
FlutterDriver does not exist in Flutter 2.5....
So we can use the above method.
any substitute for that method with WidgetTester?
